### PR TITLE
Add codespell support (config, workflow to detect/not fix) and make it fix few typos

### PR DESCRIPTION
**Pull request by yarikoptic**
More about codespell: https://github.com/codespell-project/codespell .

I personally introduced it to dozens if not hundreds of projects already and so far only positive feedback.

CI workflow has 'permissions' set only to 'read' so also should be safe.

### CODE REVIEW

File: .github/workflows/codespell.yml
Comment (skshetry) [inline .github/workflows/codespell.yml:1]: We already run pre-commit as part of our CI. So we don't need to run it separately.

https://github.com/fsspec/sshfs/blob/34b52b241d197a83c085c77ca104888e812283d9/.github/workflows/pre-commit.yml#L16

File: pyproject.toml [hunk 1]
```diff
@@ -14,3 +14,10 @@ force_grid_wrap=0
 include_trailing_comma=true
 multi_line_output=3
 use_parentheses=true
+
+[tool.codespell]
+# Ref: https://github.com/codespell-project/codespell#using-a-config-file
+skip = '.git*'
+check-hidden = true
+# ignore-regex = ''
+# ignore-words-list = ''
```
Comment (skshetry) [inline pyproject.toml:23]: I don't think we need a config _yet_. Can be added in the future if needed.

Comment (yarikoptic) [reply pyproject.toml]: ok, removed. Note that few dotfiles/directories you have would not be checked... minor sacrifice ;)

File: .github/workflows/codespell.yml
Comment (yarikoptic) [reply .github/workflows/codespell.yml:1]: removed now

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -17,3 +17,10 @@ repos:
         additional_dependencies:
         - mdformat-black
         - mdformat_tables
+- repo: https://github.com/codespell-project/codespell
+  # Configuration for codespell is in pyproject.toml
```
Comment (skshetry) [inline .pre-commit-config.yaml:21]:
```suggestion
```

**Review by skshetry — APPROVED** [review]

**Pull request merged**